src/sources.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::Result;
+
+use native_tls::Certificate;
+
+use serde::Deserialize;
+
+use std::collections::{HashMap, HashSet};
+use std::path::PathBuf;
+use std::{fs, fs::File};
+
+#[derive(Default, Deserialize, Debug)]
+pub(crate) struct Sources {
+    #[serde(default)]
+    insecure_sources: HashSet<String>,
+    #[serde(default)]
+    source_authorities: HashMap<String, CertificateAuthority>,
```

Comment:
nitpicking, you can use the `#[serde(default)]` at the container nevel (aka above `struct` declaration)